### ISSUE TITLE
Number of sockets grows

### ISSUE THREAD

**Issue by tlienart** [asprionj/LiveServer.jl#3]
So currently, the `ws_tracker` function 

* creates a new websocket
* appends it to the list of websocket associated with one file

But that seems wasteful, in practice for every upgrade request, new websockets are added. In practice, if you display the number of websockets associated with one file, it just grows and grows and grows even though there is some filtering for active/open sockets.

**Edit**: to see this, print the number of sockets at every upgrade request and hit save a couple of time (even without changes) in `index.html` it increments by 1 each time.

It seems to me that we should be able to detect if the upgrade request comes from a place we've seen before, in which case there shouldn't be a new websocket but it should use the existing one (?) unless it has disappeared. 

What do you think?

**Comment by tlienart** [comment]
Ok so a possible approach (maybe not the best) would be that, upon message, the script respondes with the tab id. Something like

```javascript
ws_M3sp9eAgRFN9y.onmessage = function(msg) {
          ws_M3sp9eAgRFN9y.send(browser.tabs.getCurrent().id)
          ws_M3sp9eAgRFN9y.close();
          location.reload();
      };
```

this is a unique ID for the tab the script is ran from which would allow us to not reopen a websocket if it's already there. The only thing required (that I don't know how to do yet) is how to recuperate that message appropriately so that it can be processed in Julia.

**Comment by asprionj** [comment]
Just had an idea during a walk with my dog ;-)

Actually, when we send the message from the server, it has one single purpose: trigger the client to reload the page. So, in any case, the server can close and forget the WS-connection after sending the message:

- Client already disconnected: ok, so it's the right thing to remove the connection also on the server.
- Client does not react to message: there's something wrong with the client, so no need to send further messages anyway.
- Client does receive message and acts as it should: client will start a new connection from the re-loaded page anyway, so the old one is closed (or orphaned, if the client-side `close()` doesn't do it's job).

**Comment by tlienart** [comment]
Yeah that makes sense 👍

**Comment by tlienart** [comment]
I'm re-opening not because it's not fixed but because if we use #7 then this change should be ported there too. Will close after that
